### PR TITLE
Documentation

### PR DESCRIPTION
**Pull request by naomijub**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,7 +1,242 @@
 # Transistor
 
-A Crux Client crate.
+A Crux Client crate. For now, this crate intends to support 2 ways to interact with Crux:
 
-For examples on usage, please refer to [examples directory](https://github.com/naomijub/transistor/tree/master/examples), and for information on crux and how to use it, please follow the link to [opencrux](https://opencrux.com/docs#restapi)
+* Via `Docker` with a [`crux-standalone`](https://opencrux.com/docs#config-docker) version [docker-hub](https://hub.docker.com/r/juxt/crux-standalone).
```
Comment (otaviopace) [inline README.md:5]: What do you think of those two items being checkboxes? Being the first one checked and the second one unchecked.

File: README.md [hunk 2]
```diff
@@ -1,7 +1,242 @@
...
+* **TODO:** Via [`HTTP`](https://opencrux.com/docs#config-http) using the [`RESTapi`](https://opencrux.com/docs#restapi).
```
Comment (otaviopace) [inline README.md:6]: If you like the checkbox idea, we could remove the TODO here

Comment (otaviopace) [reply README.md:6]: Also, I would separate the `RESTapi` into two words, and make `api` upper case.

File: README.md [hunk 3]
```diff
@@ -1,7 +1,242 @@
...
+
+* For information on Crux and how to use it, please follow the link to [opencrux](https://opencrux.com/docs#restapi). Note that the current crate version (`Docker only`) uses a few modified endpoints due to its Docker implementation.
+
+* For examples on usage, please refer to [examples directory](https://github.com/naomijub/transistor/tree/master/examples).
+
+## Creating a Crux Client
+All operations with Transistor start in the module `client` with `Crux::new("localhost", "3000")`.  The struct `Crux` is responsabile for defining request `HeadersMap` and the request `URL`. The `URL` definition is required and it is done by the static function `new`, which receives as argument a `host` and a `port` and returns a `Crux` instance. To change `HeadersMap` info so that you can add `AUTHORIZATION` you can use the function `with_authorization` that receives as argument the authorization token and mutates the `Crux` instance.
+* `HeaderMap` already contains the header `Content-Type: application/edn`.
+
+Finally, to create a Crux Client the function `client` should be called. This function returns a struct that contains all possible implementarions to query Crux Docker.
```
Comment (otaviopace) [inline README.md:18]: Maybe here should be explained that the function isn't called `client`, it is a function with the `_client` suffix.

File: README.md [hunk 4]
```diff
@@ -1,7 +1,242 @@
...
+// DockerClient with AUTHORIZATION
+let auth_client = Crux::new("127.0.0.1","3000").with_authorization("my-auth-token").docker_client();
+
+// DockerClient without AUTHORIZATION
+let client = Crux::new("127.0.0.1","3000").docker_client();
+```
+
+## Docker Client
+Once you have called `docker_client` you will have an instance of the `DockerClient` struct which has a bunch of functions to query Crux on Docker:
+* `state` queries endpoint `/` with a `GET`. No args. Returns various details about the state of the database.
```
Comment (otaviopace) [inline README.md:31]: I think each one of these items (state, tx_log, etc) should have the link to the documentation, so the developer can see the `edn` and docs on the route. What do you think?

File: README.md [hunk 5]
```diff
@@ -1,7 +1,242 @@
...
+//     index___index_version: 5, 
+//     doc_log___consumer_state: None, 
+//     tx_log___consumer_state: None, 
+//     kv___kv_store: "crux.kv.rocksdb.RocksKv", 
+//     kv___estimate_num_keys: 56, 
+//     kv___size: 2271042 
+// }
+```
+
+* `tx_log` requests endpoint `/tx-log` via `POST`. A Vector of `Action` is expected as argument. The "write" endpoint, to post transactions..
```
Comment (otaviopace) [inline README.md:45]: There's two dots on the end of the phrase.

File: README.md [hunk 6]
```diff
@@ -1,7 +1,242 @@
...
+// }
+```
+
+`Action` is an enum with a set of options to use in association with the function `tx_log`:
+* [`PUT`](https://opencrux.com/docs#transactions-put) - Write a version of a document
+* [`Delete`](https://opencrux.com/docs#transactions-delete) - Deletes the specific document at a given valid time
+* [`Evict`](https://opencrux.com/docs#transactions-evict) - Evicts a document entirely, including all historical versions (receives only the ID to evict)
+
+## Dependencies
+A strong dependency of this crate is the [edn-rs](https://crates.io/crates/edn-rs) crate, as many of the return types are in the [Edn format](https://github.com/edn-format/edn). The sync http client is `reqwest` with the feature `blocking`.
```
Comment (otaviopace) [inline README.md:242]: What do you think of changing the order of `feature` and `blocking` on the end of the line? Idk if this is nitty picking, if you think it is good as it is, just dismiss the comment 🙂

File: src/client.rs [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+    }
+
+    #[cfg(test)]
+    fn uri(&self) -> String {
+        use mockito::server_url;
+        server_url()
+    }
+
+    /// To query database on Docker via http it is necessary to use `DockerClient` 
+    pub fn docker_client(&mut self) -> DockerClient {
```
Comment (otaviopace) [inline src/client.rs:40]: Thinking now, knowing that Crux has multiple ways of being consumed, this `_client` method really become handy and it seems like a smart design decision 👏 

You can dismiss this, I just wanted to say your idea is pretty good 😊

File: README.md (hunk 4 above)
Comment (naomijub) [reply README.md:31]: Yes, will do this after this changes have been updated

Comment (naomijub) [reply README.md:32]: https://github.com/naomijub/transistor/issues/15

File: src/client.rs [hunk 8]
```diff
@@ -0,0 +1,100 @@
+use reqwest::{
+    header::{HeaderMap,AUTHORIZATION, CONTENT_TYPE},
+};
+
```
Comment (diegoferreirarj) [inline src/client.rs:4]: request::

Comment (naomijub) [reply src/client.rs:4]: the name of the crate is `reqwest` kkkkkk

**Pull request merged**